### PR TITLE
Dcam run logic

### PR DESCRIPTION
**Pull request by omarjamil**
This corrects some errors in camera trigger logic. There are some placeholder functions for the experiment logic that need to be hooked up to the frontend. This merge can be built on for the logic implementation.

### CODE REVIEW

**Review by felicityguest — CHANGES_REQUESTED** [review]
I've made a few comments, most of which aren't important. Two comments made in relation to `experiment_loop` and `image` in `netclamp/camera/hamamatsu.py` might be though.

To clarify, I'm not necessarily requesting changes, just think a couple of checks are needed prior to merging.

File: netclamp/session/session.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class SessionInfo:
+    _shared_state = {'save_path':None, 'frame_save_filename':None, 'filters':[]}
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply netclamp/session/session.py]: Changed

File: netclamp/camera/hamamatsu.py [hunk 2]
```diff
@@ -5,13 +5,20 @@
...
     """
     Hamamatsu camera.
     """
 
+    # def __init__(self):
+    #     pass
+
+    # def capture_image(self):
+    #     pass
+
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply netclamp/camera/hamamatsu.py:21]: Yes, once the testing is done, we can remove these. At the moment need these if trying to do anything with hamamatsu.py file not on Windows.

File: netclamp/camera/hamamatsu.py [hunk 3]
```diff
@@ -29,12 +36,14 @@ def __init__(self):
         self.capture = self._dcam[0]
         self.capture.open()
 
-        self.set("trigger_mode", "external")
+        self.experimental_loop = False
+        #self.set("trigger_source", "external")
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply netclamp/camera/hamamatsu.py]: Removed

File: netclamp/camera/hamamatsu.py [hunk 4]
```diff
@@ -46,11 +55,43 @@ def capture_image(self) -> np.ndarray:
...
+        """
+        As the experiment enters trigger loop, start capturing the
+        images when ready. Instead of returning the images, this will
+        save the images
+        """
+        self.experimental_loop = True
+        self.set("trigger_source","external")
+        ROI = RegionOfInterest()
+        ROI.create_rois_from_file()
+        while(experiment_loop):
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply netclamp/camera/hamamatsu.py]: Changed all of them to `experiment_loop`. Also removed the brackets and changed to `self.experiment_loop`.

File: netclamp/camera/hamamatsu.py [hunk 5]
```diff
@@ -46,11 +55,43 @@ def capture_image(self) -> np.ndarray:
...
+            with dcam.Stream(self.capture, 1) as stream:
+                self.capture.start(live=False)
+                frame = dcam.copy_frame(next(stream))
+                self.capture.stop()
+                image = ROI.process(frame)
+                self.save_frame(frame, save_path, save_name)
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply netclamp/camera/hamamatsu.py:86]: I don't think we need to be saving all of the frames, but need to work through that with Kyle. It is likely only the flux time series for each of the ROIs that we need to save.

File: app/api/camera.py [hunk 6]
```diff
@@ -3,13 +3,14 @@
 import os
 
 from netclamp import camera
-
+from netclamp import session
 from . import settings
 from .session import Session
```
[the remark this answers is not in the document]
Comment (omarjamil) [reply app/api/camera.py:8]: It is a bit confusing, but that is the way to pass any session information to the camera, though at the moment not passing anything.

**Pull request merged**